### PR TITLE
Strays 'nothing' type for empty tuple inference.

### LINKED ISSUE #549: Stray 'nothing' type for empty tuple inference
While using `merge-pyi -i`, an unannotated function that returns an empty tuple got as type `Tuple[nothing, ...]`. Since this type is not defined, a later pytype run fails. If the function is already annotated, nothing happens.

### Input file:

```
from typing import Tuple

def empty_tuple1():
  return ()

def empty_tuple2() -> Tuple:
  return ()
```

### Output pyi
```
# (generated with --quick)

from typing import Tuple

def empty_tuple1() -> Tuple[nothing, ...]: ...
def empty_tuple2() -> tuple: ...
```

### PR DESCRIPTION
**Pull request by souravbadami**
Fixes #549.

### CODE REVIEW

File: pytype/pytd/pytd_visitors.py [hunk 1]
```diff
@@ -694,7 +699,9 @@ def MaybeCapitalize(self, name):
   def VisitGenericType(self, node):
     """Convert a generic type to a string."""
     parameters = node.parameters
-    if self._NeedsTupleEllipsis(node):
+    if self._IsEmptyTuple(node):
+      return self.MaybeCapitalize(node.base_type) + "[()]"
```
Comment (martindemello) [inline pytype/pytd/pytd_visitors.py:703]: just set `parameters = "()"` here

Comment (martindemello) [reply pytype/pytd/pytd_visitors.py:703]: sorry, `parameters = ("()",)` since it needs to be a tuple

Comment (souravbadami) [reply pytype/pytd/pytd_visitors.py]: Oh, yeah. I could have done that. 
I need to read a bit more and get familiar with the codebase.
Resolved.

**Comment by souravbadami** [comment]
@rchen152 - Could you please review the changes?

**Comment by souravbadami** [comment]
I can't merge it as I don't have permission to do it. Thank you!

**Comment by rchen152** [comment]
Heh, that's true.

FYI the export failed because of an unrelated change that broke some tests - hopefully I'll get it sorted out on Monday.

**Pull request closed without merging**